crates/ruff_python_ast/src/identifier.rs

```diff
@@ -251,10 +228,13 @@ impl<'a> IdentifierTokenizer<'a> {
         while let Some(c) = self.cursor.bump() {
             match c {
                 c if is_python_identifier_start(c) => {
-                    let start = self.offset.add(self.cursor.offset()).sub(c.text_len());
+                    self.offset += self.cursor.token_len();
+                    self.cursor.start_token();
                     self.cursor.eat_while(is_python_identifier_continue);
-                    let end = self.offset.add(self.cursor.offset());
-                    return Some(TextRange::new(start, end));
+
+                    let token_offset = self.offset - c.text_len();
+                    let token_len = self.cursor.token_len() + c.text_len();
+                    return Some(TextRange::at(token_offset, token_len));
```

Comment:
I wonder if we should panic on any other token to avoid that the tokenizer eg returns a name from inside a string (because it happily lex over quotes)